### ISSUE TITLE
[BUG] Delete model button deletes the whole folder.

### ISSUE THREAD

**Issue by gontz** [BlafKing/sd-civitai-browser-plus#29]
**Describe the bug**
When a Lora version file has been selected for deletion, the delete button deletes the whole folder where the file is. Unfortunately, the folder is a folder where all my Loras are stored. All of them were deleted.

**Comment by BlafKing** [comment]
I'm very sorry to hear that :(
I appreciate you writing in.
That's a very big mistake on my part, I'll start working on this issue right away so that it won't happen to any else.
Apologies that my poor code made you delete all your Lora's, apologies :(

**Comment by BlafKing** [comment]
Weirdly enough I can't seem to recreate the issue, if I delete any model using the delete button it only deletes the matching model, not the entire folder.
I'm not exactly sure what could've caused this issue but I'll add an extra check to verify that it won't delete more than one file.

Could you provide more details such as your OS?

**Comment by gontz** [comment]
Yes. Windows 10. 
My Folder structure: models\Lora\SDXL\
When I downloaded the Lora with the other civitai plugin, it asked for the creation of a download folder. It was Lora\SDXL. 
I assume that the folder name was used as the file location. Instead of deleting just the safetensor file, it deleted the whole folder assuming that the folder contains only the safetensor and json and preview file

**Comment by BlafKing** [comment]
Do you perhaps know which model you tried to delete?
Currently as far as I can tell it shouldn't be possible to delete an entire folder, here's the code that gets triggered when you delete a file:
```
model_folder = os.path.join(_api.contenttype_folder(content_type))  # Get the folder path related to a content type.

path_file = None  # Initialize a variable to store the file path.

base_name = os.path.splitext(model_filename)[0]  # Extract the base name of the model filename without the extension.

for root, dirs, files in os.walk(model_folder):  # Iterate through all folders and files within model_folder.
    for file in files:  # Iterate through files in the current directory.
        file_base_name = os.path.splitext(file)[0]  # Extract the base name of the current file.
        if base_name in file_base_name:  # Check if the base name matches the target model file's base name.
            path_file = os.path.join(root, file)  # Construct the full path to the matching file.
            os.remove(path_file)  # Remove the file.
            print(f"Removed: {path_file}")  # Print a message indicating the removal.
```

I'll implement an extra check to double check that an entire folder cannot be removed, and an extra check that makes sure not more than one model is removed to make sure your issue won't happen again.

**Comment by BlafKing** [comment]
I've slightly changed the code so that it now has an extra check to verify it is indeed deleting a file not a folder.
I also changed the code to match the name it's searching for completely, instead of also deleting if the name partially matches.

Lastly I also changed the extension so that it now moves models to the Trash instead of fully deleting them.
Thanks for reporting the issue, I'll close this thread now since the issue should be fixed.
Apologies again for making you lose all your LORA's, hope this solves it for good!
